### ISSUE TITLE
Multi threading phase 3: serialize multiple columns simultaneously

### ISSUE THREAD

**Issue by fstpackage** [fstpackage/fst#64]
Some column-types can be serialized more efficiently than others. Character vector serialization is slow compared to integer, double, logical and factor serialization. We can process serialization of a character vector in parallel with the serialization of a column of another type. Combining serialization of a _slow_ column with the serialization of a _fast_ column enables maximum throughput of data and keeps the IO device busy.

In short: combine a column that can be (de-) serialized faster than IO speed with a column that can be (de-) serialized at lower than IO speed and process in parallel.
